### PR TITLE
feat(report): add Report Post Core layer

### PR DESCRIPTION
**Pull request by infin1t3**
## PR Highlights: Report Post Core Layer

Second part of #591

### Architecture

Follows Franky's layered pattern: Route → Controller → Application → Service
Clean separation of concerns with validation in Pipes layer
Extends existing Chatwoot service for report submission

### Pipes layer (`core/pipes/report/`)

**ReportValidators** — static validation methods:
- `validatePubky()` — reporter's pubky validation (required, non-empty)
- `validatePostUrl()` — post URL format validation (required, valid URL)
- `validateIssueType()` — validates against 9 allowed issue types
- `validateReason()` — presence check + max length (1000 chars)
- `validateName()` — reporter's display name validation (required, non-empty)

Each validator returns the trimmed, validated value.
Throws `AppError` with descriptive messages on failure.

### Controller layer (`core/controllers/report/`)

**ReportController.submit()** — entry point for API route
- Orchestrates validation via ReportValidators
- Collects validated inputs into a single object
- Delegates to ReportApplication for business logic
- Single responsibility: validate input, delegate execution

### Application layer (`core/application/report/`)

**ReportApplication.submit()** — business logic orchestration
- Builds source label: `Report Post - {Issue Type Label}`
- Builds comment body with post URL and user-provided reason
- Delegates to ChatwootService with `report` type
- Name and pubky are passed as contact info (not in message body)
- Error logging via `Libs.Logger.error()`

### Service layer (`core/services/chatwoot/`)

Refactored `ChatwootService.submit()` to accept `type` parameter:
- Dynamic inbox selection based on submission type:
  - `feedback` → Inbox 26
  - `report` → Inbox 27
- New constants: `CHATWOOT_INBOX_IDS`, `CHATWOOT_SUBMISSION_TYPES`
- New type: `ChatwootSubmissionType`
- Message prefix varies by type (`Feedback` vs custom source label for reports)

### API route (`app/api/report/`)

- Replaces stub with real `ReportController.submit()` call
- `AppError` handling with proper status codes
- Unexpected error logging via `Libs.Logger.error()`
- Includes GET (405) and OPTIONS handlers

### Test coverage

- **ReportValidators** — all validation scenarios (valid inputs, edge cases, error cases)
- **ReportController** — success, validation errors, application failures
- **ReportApplication** — message formatting, Chatwoot integration, error handling
- **API route** — success, AppError handling, unexpected errors, all field validations

### CODE REVIEW

File: src/core/services/chatwoot/chatwoot.constants.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * Inbox IDs and submission types for Chatwoot API integration.
+ */
+
+/**
+ * Chatwoot Inbox IDs
+ *
+ * Different inboxes for different submission types.
+ * These map to Chatwoot inbox configurations.
+ */
+export const CHATWOOT_INBOX_IDS = {
```
Comment (infin1t3) [inline src/core/services/chatwoot/chatwoot.constants.ts]: I am not sure should this be in an env file or just as a constant in code.
